sound/soc/sof/ipc3.c

```diff
@@ -148,6 +149,8 @@ static void ipc3_log_header(struct device *dev, u8 *text, u32 cmd)
 		case SOF_IPC_TRACE_DMA_PARAMS:
 			str2 = "DMA_PARAMS"; break;
 		case SOF_IPC_TRACE_DMA_POSITION:
+			if (sof_debug_check_flag(flags))
```

Comment:
Better to use the flag itself here so we don't need to look back what it is?

Reply:
I kind of liked this noise as we can see that new data is coming in via dtrace from firmware. If the sof-logger came back with empty log we can see in the dmesg if we have aver received the position update.
I still have the two dtrace PRs open to print the new positions... (#3516)

Reply:
you still have this capability, it's just not there by default. I think we have to separate out debugging the trace and debugging firmware, if we have all logs all the time it just makes it harder for everyone, and we have already too many logs.

Reply:
> Better to use the flag itself here so we don't need to look back what it is?

I used this to avoid a 100+ line. I can move the definition down but it'll look ugly. I don't have a good solution.

Reply:
direct use would put it to 92, but surely it looks a bit out of place, but not too much.
```c
	switch (glb) {

	case SOF_IPC_GLB_TRACE_MSG:
		str = "GLB_TRACE_MSG";
		switch (type) {
		case SOF_IPC_TRACE_DMA_PARAMS:
			str2 = "DMA_PARAMS"; break;
		case SOF_IPC_TRACE_DMA_POSITION:
			if (sof_debug_check_flag(SOF_DBG_SQUELCH_DMA_POSITION_UPDATE_LOGS))
				return;
			str2 = "DMA_POSITION"; break;
		case SOF_IPC_TRACE_DMA_PARAMS_EXT:
			str2 = "DMA_PARAMS_EXT"; break;
		case SOF_IPC_TRACE_FILTER_UPDATE:
			str2 = "FILTER_UPDATE"; break;
		case SOF_IPC_TRACE_DMA_FREE:
			str2 = "DMA_FREE"; break;
		default:
			str2 = "unknown type"; break;
		}
		break;
```

Reply:
Opt-in version of the flag would be 93